File: Frontend/src/interfaces/v12/warehouse/warehouse.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface WarehouseDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface WarehouseCreateUpdateDto {
  name: string;
}

export interface WarehouseSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/Warehouse/Warehouse.cs
 using Wms.Api.Entities; 

namespace Wms.Api.Dto.Warehouse
{
    public class WarehouseDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class WarehouseCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class WarehouseSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}
